internal/assemble/assemble.go

```diff
@@ -0,0 +1,75 @@
+package assemble
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/canonical/sqlair/internal/parse"
+	"github.com/canonical/sqlair/internal/typeinfo"
+)
+
+// AssembledExpr represents an SQL expression after the input and output parts
+// have been replaced by their corresponding expressions.
+type AssembledExpr struct {
+	ParsedExpr *parse.ParsedExpr
+	SQL        string
+}
+
+type typeNameToInfo map[string]*typeinfo.Info
+
+// assembleInput prepares an input part.
```

Comment:
If I understand it correctly, this method checks whether  the `Source` in `p` has a valid Prefix first, and if so then checks whether there's a `Field` entry with name `Source.Name`. It doesn't return the Field. Feels like it's a validation method, working on the `Source`. That makes me think, should the method name be altered accordingly? (and the docstring?)